src/components/Tables/DomainTable/DomainTable.tsx

```diff
@@ -203,56 +202,69 @@ const DomainTable: React.FC<DomainTableProps> = ({
 					);
 				}
 
+				const loaded = data;
+
 				loaded.push({
 					domain: domain,
 					metadata: metadata,
 					bids: filteredBids || bids || [],
 				});
 
-				if (completed === count) {
-					setLoadedDomains(loaded);
-					setHasMetadataLoaded(true);
-					setIsLoading(false);
-					if (onLoad) {
-						onLoad();
-					}
-				}
+				setData(loaded);
+
+
 			} catch (e) {}
 		};
 
-		for (let i = 0; i < domains.length; i++) {
-			if (!domains[i].metadata) continue;
-			count++;
-			getData(domains[i]);
-		}
-
-		if (!count) {
-			setHasMetadataLoaded(true);
-			setIsLoading(false);
+		if (isSubscribed === false && domains.length > 0) {
+			//if finished before call to all asyncs fetches, and the domains exist
+			setIsSubscribed(true);
+			for (let i = 0; i < domains.length; i++) {
+				if (!domains[i].metadata) continue;
+				getData(domains[i]);
+			}
 		}
 	}, [domains]);
 
 	useEffect(() => {
 		if (!isLoading && onLoad) onLoad();
 	}, [isLoading]);
 
+	/////////////////////
+	///// Functions /////
+	/////////////////////
+
+	const feed = () => {
```

Comment:
Where/when does feed get called?

Reply:
That function is for Brett, he needs to test it with UI changes before continue